prime-angular-frontend/src/app/modules/enrolment/shared/guards/enrolment.guard.ts

```diff
@@ -302,7 +302,7 @@ export class EnrolmentGuard extends BaseGuard {
   private onInitialEnrolmentCheckForMatchingPaperEnrollee(): UnaryFunction<Observable<[string, Enrolment]>, Observable<Enrolment>> {
     return pipe(
       exhaustMap(([dateOfBirth, enrolment]: [string, Enrolment]) =>
-        (this.enrolmentService.isInitialEnrolment && dateOfBirth && this.enrolmentService.isMatchingPaperEnrollee === null)
+        (this.enrolmentService.isInitialEnrolment && dateOfBirth && this.enrolmentService.isMatchingPaperEnrollee === null && !enrolment?.approvedDate)
```

Comment:
Would `approvedDate` ever not be a key on the model?

Reply:
No, but on first login enrolment is undefined